torchmetrics/text/rouge.py

```diff
@@ -0,0 +1,144 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from torch import Tensor
+
+from torchmetrics import Metric
+from torchmetrics.functional.text.rouge import _rouge_score_compute, _rouge_score_update
+from torchmetrics.utilities.imports import _NLTK_AVAILABLE, _ROUGE_SCORE_AVAILABLE
+
+if _ROUGE_SCORE_AVAILABLE:
+    from rouge_score.rouge_scorer import RougeScorer
+    from rouge_score.scoring import BootstrapAggregator
+else:
+    RougeScorer, BootstrapAggregator = object, object
+
+
+class ROUGEScore(Metric):
+    """
+    Calculate `ROUGE score <https://en.wikipedia.org/wiki/ROUGE_(metric)>`_, used for automatic summarization.
+
+    Args:
+        preds:
+            An iterable of predicted sentences.
+        targets:
+            An iterable of target sentences.
```

Comment:
as preds and targets are not input to the `__init__` function lets re-organise this
```suggestion
    Forward accepts:
        - preds: An iterable of predicted sentences.
        - targets: An iterable of target sentences.

    Args:     
```

Reply:
Should I just omit them as I have already mentioned it in the docstring of the `update` method ?